internal/controllers/preflightvalidation_reconciler.go

```diff
@@ -230,10 +239,12 @@ func (p *preflightReconcilerHelperImpl) processPreflightValidation(ctx context.C
 			RegistryTLS:   mod.RegistryTLS,
 		}
 		micName := mod.Name + "-preflight"
-		err := p.micAPI.CreateOrPatch(ctx, micName, mod.Namespace, []kmmv1beta1.ModuleImageSpec{micObjSpec}, mod.ImageRepoSecret, mod.ImagePullPolicy, pv)
+		// We only care about the last opRes as we want to make sure we give enough time for the MIC to be created
+		opRes, err = p.micAPI.CreateOrPatch(ctx, micName, mod.Namespace, []kmmv1beta1.ModuleImageSpec{micObjSpec},
```

Comment:
what if the first opRes is "Created" and the last is "Patch"?

Reply:
Then we return "Patch" and ignore it in the caller. By this time, we know for sure that it was already created cause we had time to patch it.